utils/http/request.go

```diff
@@ -0,0 +1,156 @@
+package http
+
+import (
+	"fmt"
+	"strings"
+
+	validation "github.com/go-ozzo/ozzo-validation/v4"
+	"github.com/go-ozzo/ozzo-validation/v4/is"
+
+	configUtils "github.com/ARM-software/golang-utils/utils/config"
+)
+
+const (
+	AuthorisationSchemeToken        = "Token"
+	AuthorisationSchemeBasic        = "Basic"
+	AuthorisationSchemeBearer       = "Bearer"
+	AuthorisationSchemeConcealed    = "Concealed"
+	AuthorisationSchemeDigest       = "Digest"
+	AuthorisationSchemeDPoP         = "DPoP"
+	AuthorisationSchemeGNAP         = "GNAP"
+	AuthorisationSchemeHOBA         = "HOBA"
+	AuthorisationSchemeMutual       = "Mutual"
+	AuthorisationSchemeNegotiate    = "Negotiate"
+	AuthorisationSchemeOAuth        = "OAuth"
+	AuthorisationSchemePrivateToken = "PrivateToken"
+	AuthorisationSchemeSCRAMSSHA1   = "SCRAM-SHA-1"
+	AuthorisationSchemeSCRAMSHA256  = "SCRAM-SHA-256"
+	AuthorisationSchemeVapid        = "vapid"
+)
+
+var (
+	// HTTPAuthorisationSchemes lists all supported authorisation schemes. See https://www.iana.org/assignments/http-authschemes/http-authschemes.xhtml
+	HTTPAuthorisationSchemes = []string{
+		AuthorisationSchemeToken,
+		AuthorisationSchemeBasic,
+		AuthorisationSchemeBearer,
+		AuthorisationSchemeConcealed,
+		AuthorisationSchemeDigest,
+		AuthorisationSchemeDPoP,
+		AuthorisationSchemeGNAP,
+		AuthorisationSchemeHOBA,
+		AuthorisationSchemeMutual,
+		AuthorisationSchemeNegotiate,
+		AuthorisationSchemeOAuth,
+		AuthorisationSchemePrivateToken,
+		AuthorisationSchemeSCRAMSSHA1,
+		AuthorisationSchemeSCRAMSHA256,
+		AuthorisationSchemeVapid,
+	}
+	inAuthSchemes = []any{
+		AuthorisationSchemeToken,
+		AuthorisationSchemeBasic,
+		AuthorisationSchemeBearer,
+		AuthorisationSchemeConcealed,
+		AuthorisationSchemeDigest,
+		AuthorisationSchemeDPoP,
+		AuthorisationSchemeGNAP,
+		AuthorisationSchemeHOBA,
+		AuthorisationSchemeMutual,
+		AuthorisationSchemeNegotiate,
+		AuthorisationSchemeOAuth,
+		AuthorisationSchemePrivateToken,
+		AuthorisationSchemeSCRAMSSHA1,
+		AuthorisationSchemeSCRAMSHA256,
+		AuthorisationSchemeVapid,
+	}
+)
+
+// Auth defines a typical HTTP client authentication/authorisation configuration
+// See https://datatracker.ietf.org/doc/html/rfc7235
+type Auth struct {
+	Enforced    bool   `mapstructure:"enforced"`
+	Scheme      string `mapstructure:"scheme"`
+	AccessToken string `mapstructure:"token"`
+}
+
+func (cfg *Auth) GetAuthorizationHeader() string {
+	if cfg.Enforced {
+		return fmt.Sprintf("%v %v", cfg.Scheme, cfg.AccessToken)
+	}
+	return ""
+}
```

Comment:
If you call this on an enforced but non-validated header (e.g. it has the scheme set but not the token and no-one called Validate()) you could end up with a hard to debug error as it might return a header that gets parsed quite far through the stack before having issues and you could get an error with just a blank string where it isn't clear what the issue is.

I know it isn't super nice but I think the chance of errors due to missing a call to validate is too likely. I would suggest something like:
```suggestion
const (
    missingScheme      = "!!MISSING_SCHEME!!"
    missingAccessToken = "!!MISSING_ACCESS_TOKEN!!"
)

func (cfg *Auth) GetAuthorizationHeader() string {
	if !cfg.Enforced {
	    return ""
	}
	
	scheme, accessToken := cfg.Scheme, cfg.AccessToken
	if scheme == "" {
	    scheme = missingScheme
	}
	if accessToken == "" {
	    accessToken = missingAccessToken
	}
	    
    return fmt.Sprintf("%v %v", cfg.Scheme, cfg.AccessToken)
}
```

Then if it is misused it is more obvious what the error is